backend/layers/business/business.py

```diff
@@ -83,6 +86,32 @@ def __init__(
         self.uri_provider = uri_provider
         super().__init__()
 
+    @staticmethod
+    def _get_published_at_and_collection_version_id_else_not_found(
```

Comment:
also, might be worth renaming to be explicit that this is for fetching a dataset version's published_at

Reply:
since published_at and collection_version_id may, at first glance, suggest this is about collection version fields